### ISSUE TITLE
Clean install fails due to numpy incompatibility with Python 3.9

### ISSUE THREAD

**Issue by sezelt** [py4dstem/py4DSTEM#134]
When running a clean install in a fresh conda environment, various pip errors arise due to the unavailability of binary wheels for numpy and scipy (cf., https://github.com/numpy/numpy/issues/17569, https://numpy.org/news/, my error pasted below) for Python 3.9. The numpy page recommends installing with the `--only-binary=:all:` option passed to `pip` but this did not work for me, as it seems to produce too many version conflicts and the environment can't be resolved. Installing numpy and scipy via through conda before running `pip install py4dstem` also leads to errors as it still tries to build numpy from source.

It seems that forcing python 3.7 (or maybe 3.8, which I haven't tested) is needed until the scientific python packages catch up to 3.9:

```bash
conda update conda
conda create -n py4dstem python=3.7
conda activate py4dstem
conda install pip
pip install py4dstem
```
I've tested that the above commands correctly install py4DSTEM in a fresh environment on my iMac.

---------------

#### `pip` error
```
...
Collecting scikit-image>=0.16.2
  Using cached scikit-image-0.17.2.tar.gz (29.8 MB)
    ERROR: Command errored out with exit status 1:
     command: /Users/sez/opt/miniconda3/envs/py4D-39/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/ts/tq6v7mks7hvg37ys5zvs1c2w0000gn/T/pip-install-8i0jj9cc/scikit-image_41616947421f43f3885c8dcd5d3fe4ec/setup.py'"'"'; __file__='"'"'/private/var/folders/ts/tq6v7mks7hvg37ys5zvs1c2w0000gn/T/pip-install-8i0jj9cc/scikit-image_41616947421f43f3885c8dcd5d3fe4ec/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/ts/tq6v7mks7hvg37ys5zvs1c2w0000gn/T/pip-pip-egg-info-ku2h9qn4
         cwd: /private/var/folders/ts/tq6v7mks7hvg37ys5zvs1c2w0000gn/T/pip-install-8i0jj9cc/scikit-image_41616947421f43f3885c8dcd5d3fe4ec/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/ts/tq6v7mks7hvg37ys5zvs1c2w0000gn/T/pip-install-8i0jj9cc/scikit-image_41616947421f43f3885c8dcd5d3fe4ec/setup.py", line 234, in <module>
        'build_ext': openmp_build_ext(),
      File "/private/var/folders/ts/tq6v7mks7hvg37ys5zvs1c2w0000gn/T/pip-install-8i0jj9cc/scikit-image_41616947421f43f3885c8dcd5d3fe4ec/setup.py", line 58, in openmp_build_ext
        from numpy.distutils.command.build_ext import build_ext
    ModuleNotFoundError: No module named 'numpy'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by sdlszst** [comment]
Thanks Steve, I have the same error too, and forcing python 3.7 works!

**Comment by w4m0330** [comment]
I had the same problem doing a fresh install of py4dstem with Python 3.9. 
I force installed it with Python 3.6, but encountered incompatibility issue with ncenmpy. 
Now I tried it with Python 3.8.5, it worked for me (windows)

**Comment by ercius** [comment]
ncempy 1.8.1 fixes the issue for python 3.6 and is pushed to pypi.

**Comment by bsavitzky** [comment]
Thanks Steve, and others - the install instructions now specify python 3.7.  It seems that 3.8 works in general as well, but threw a weird pip error on one of my checks on a mac, so sticking with 3.7 for now.
